### ISSUE TITLE
Record accessors not exporting in sub-modules

### ISSUE THREAD

**Issue by eayus** [idris-lang/Idris2#2248]
Sub modules do not get automatic access to record accessors.

For example, in a module hierarchy `A` with submodule `A.B`:

```idris
module A

export
record R where
    constructor C
    x : Int
```
```idris
module A.B

import A

-- Compiles fine
f : R -> Int
f (C x) = x

-- Error! (.x) is not accessible
g : R -> Int
g r = r.x
```
The submodule is allowed to pattern match on the type `R` and extract fields that way, but not use the record accessors directly.

**Comment by andorp** [comment]
Would `public export` work?

**Comment by eayus** [comment]
@andorp
Yes it would, however my understanding is that sub-modules should get full access to parent modules' definitions regardless of the export modifier.

**Comment by andorp** [comment]
I see. It is not clear to me, how we consider sub-modules. My vague mental model is that namespaces are different from modules somehow.

This snippet seems to be compiling.

```
record X where
  constructor MkX
  x : Int

namespace Y

  export
  f : X -> Int
  f r = r.x
```

**Comment by eayus** [comment]
I'm pretty sure namespaces are different to modules.

Submodules of `A` would just be `A.X`, `A.Y`, `A.X.P`, etc. - so any module beginning with `A`.

**Comment by gallais** [comment]
Accessors for a record are in a sub-namespace named after the record itself.
So `x` will be available in `A.R.B` but not `A.B`.
